prefect_dask/task_runners.py

```diff
@@ -209,11 +210,18 @@ async def submit(
         # where possible to optimize Dask task scheduling
         call_kwargs = self._optimize_futures(call.keywords)
 
+        if "task_run" in call_kwargs:
+            task_run = call_kwargs["task_run"]
+            flow_run = FlowRunContext.get().flow_run
+            dask_key = f"{task_run.name}-{task_run.id.hex}-{flow_run.run_count}"
+        else:
+            dask_key = key
+
         self._dask_futures[key] = self._client.submit(
             call.func,
             # Dask displays the text up to the first '-' as the name, the task run key
             # should include the task run name for readability in the dask console.
-            key=key,
+            key=dask_key,
```

Comment:
Can you move this comment up to the section above and include a note re the need for the flow run count?